### ISSUE TITLE
feature - improve linking to fragments

### ISSUE THREAD

**Issue by andrewfowlie** [dginev/ar5iv#92]
First, this is a really cool project. I've been having a look at the outputs created and they usually look great. 

One slightly hidden current feature I really like is that we can hyperlink paragraphs, equations, figures and sections. This can help streamline discussions, referee reports, and I can even imagine referring to such a URL inside a paper.  I guess this was possibly with PDFs too, but much less appealing. 

E.g., for this paper, https://ar5iv.labs.arxiv.org/html/1906.08684, I found I can add

- I can link sections using the fragment id `S{section number}` for a section,`S{section number}.SS{subsection number}`  for a subsection etc, e.g. https://ar5iv.labs.arxiv.org/html/1906.08684#S1 and https://ar5iv.labs.arxiv.org/html/1906.08684#S2.SS2.SSS2

- Paragraphs by adding `p{paragraph number}` to the complete section information, e.g. https://ar5iv.labs.arxiv.org/html/1906.08684#S2.SS2.SSS2.p2

- Equations by adding the `E{equation number}` to the section, e.g., https://ar5iv.labs.arxiv.org/html/1906.08684#S2.E7

-  Abstract by `id1.id1`, e.g. https://ar5iv.labs.arxiv.org/html/1906.08684#id1.id1

- Bibliography by `bib`, e.g. https://ar5iv.labs.arxiv.org/html/1906.08684#bib and particular items by `bib.bib{reference number}, e.g. https://ar5iv.labs.arxiv.org/html/1906.08684#bib.bib4

- Footnotes by `footnote{number}` e.g.  https://ar5iv.labs.arxiv.org/html/1906.08684#footnote2

- It appears that we can also link to parts of paragraphs, e.g. https://ar5iv.labs.arxiv.org/html/1906.08684#S2.SS2.SSS1.p1.3 gives the third part of that paragraph, and even inline math, e.g. https://ar5iv.labs.arxiv.org/html/1906.08684#S3.SS3.SSS2.p1.2.m1.2.2 but the fragment tags get quite complicated

Anyway, the feature request is that I wonder if there was a simply way to facilitate easier creation and sharing of these kinds of links. For example, if every equation number, section/subsection ... title, perhaps even the first word of each paragraph, etc, was a hyperlink to its tag, people could easily copy the links.

**Comment by dginev** [comment]
Very reasonable upgrade request.

In terms of user experience -- do you personally enjoy the github link-on-hover feature in github's markdown previews @andrewfowlie ? Say you go to the [README](README.md) here and hover over the ar5iv heading, you should see:

![image](https://user-images.githubusercontent.com/348975/155346619-1e884e32-a558-4501-8cdb-7e94e733a901.png)


That little "link symbol emoji" 🔗 can then be clicked and the link to the fragment in question gets copied to your clipboard for reuse in any other application.

I'm personally fond of it, and I think it would be quite easy to provide another small piece of javascript that enables it.

A good place to note that before we set onto the path of building a "high quality viewer application" for arXiv, there are some preliminaries that need to be taken care of with setting up a proper javascript workspace -- that is likely the reasonable bit to do next, before I pile together too many separate independent mini-scripts.

**Comment by andrewfowlie** [comment]
Yes! That system works well and is a perfect example of the kind of problem and solution I had in mind.
